src/Helper.ts

```diff
@@ -423,9 +423,8 @@ class Helper {
       // nor the entity itself, neither the entity's linked device (if any) is linked to any area.
       ? !entity.area_id && (this.areaDeviceIds.includes(entity.device_id ?? "") || !entity.device_id)
       // Area is a hass entity;
-      // The entity's linked device or the entity itself is linked to the given area.
-      : this.areaDeviceIds.includes(entity.device_id ?? "") || entity.area_id === this.area.area_id;
-
+      // The entity's linked device (and entity area id is null) or the entity itself is linked to the given area.
+      : (this.areaDeviceIds.includes(entity.device_id ?? "") && (entity.area_id === null)) || entity.area_id === this.area.area_id;
```

Comment:
```suggestion
      // The entity itself or the entity's device is linked to the given area.
      // Note: entity.area_id is set to null when using device's area.
      : entity.area_id === this.area.area_id || (!entity.area_id && this.areaDeviceIds.includes(entity.device_id ?? ""));

```

Please apply and tryout the suggestion.
Add a new commit with the suggested code if it works.

Reply:
@DigiLive will do. Would you prefer i force push a single adjusted commit to my branch, or append another commit to what's already there?

Reply:
This branch was behind the current `main` branch.
I've merged the `main` branch into this one.

Can you confirm this branch still works as expected?
Don't worry about the commits. We'll take care of them when updating the `main` branch.